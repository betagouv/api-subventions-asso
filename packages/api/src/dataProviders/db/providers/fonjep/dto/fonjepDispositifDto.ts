export default interface FonjepDispositifDto {
    ID: number;
    Libelle: string;
    FinanceurCode: string;
}
