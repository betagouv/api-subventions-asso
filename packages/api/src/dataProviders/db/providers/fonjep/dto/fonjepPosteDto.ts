export default interface FonjepPosteDto {
    [key: string]: any;
    // dates are excel dates (i.e. days since 1900-01-01)
    Code: string | null;
    DispositifId: number | null;
    PstStatutPosteLibelle: string | null;
    PstRaisonStatutLibelle: string | null;
    FinanceurPrincipalCode: string | null;
    FinanceurAttributeurCode: string | null;
    AssociationBeneficiaireCode: string | null;
    AssociationImplantationCode: string | null;
    Annee: number | null;
    MontantSubvention: number | null;
    DateFinTriennalite: number | null;
    PstTypePosteCode: string | null;
    PleinTemps: string | null;
    DoublementUniteCompte: string | null;
}
