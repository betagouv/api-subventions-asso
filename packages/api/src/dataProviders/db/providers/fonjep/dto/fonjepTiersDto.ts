export default interface FonjepTiersDto {
    [key: string]: any;

    Code: string | null;
    RaisonSociale: string | null;
    EstAssociation: string | null;
    EstCoFinanceurPostes: string | null;
    EstFinanceurPostes: string | null;
    SiretOuRidet: string | null;
    CodePostal: string | null;
    Ville: string | null;
    ContactEmail: string | null;
}
