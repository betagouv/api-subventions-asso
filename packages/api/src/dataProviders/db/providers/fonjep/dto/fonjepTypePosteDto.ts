export default interface FonjepTypePosteDto {
    Code: string | null;
    Libelle: string | null;
}
