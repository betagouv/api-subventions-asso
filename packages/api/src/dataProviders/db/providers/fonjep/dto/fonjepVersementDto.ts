// null represents an empty cell in the file
export default interface FonjepVersementDtoWithExcelDate {
    PosteCode: string;
    PeriodeDebut: number | null; // dates in excel dates (i.e. days since 1900-01-01)
    PeriodeFin: number | null; // dates in excel dates (i.e. days since 1900-01-01)
    DateVersement: number | null; // dates in excel dates (i.e. days since 1900-01-01)
    MontantAPayer: number | null;
    MontantPaye: number | null;
}

export interface FonjepVersementDto
    extends Omit<FonjepVersementDtoWithExcelDate, "DateVersement" | "PeriodeDebut" | "PeriodeFin"> {
    PeriodeDebut: Date | null;
    PeriodeFin: Date | null;
    DateVersement: Date | null;
}
