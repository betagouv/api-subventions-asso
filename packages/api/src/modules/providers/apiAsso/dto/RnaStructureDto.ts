export interface RnaStructureDto {
    identite: {
        nom: string;
        sigle: string;
        id_rna: string;
        date_pub_jo: string;
        date_modif_rna: string;
        active: boolean;
        nature: string;
    };
    activites: {
        objet: string;
        lib_objet_social1: string;
        annee_activite_principale: number;
        effectif_salarie_cent: number;
        annee_effectif_salarie_cent: number;
    };
    coordonnees: {
        adresse_siege: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            cp: number;
            bp?: string;
            type_voie: string;
        };
        adresse_gestion: {
            cplt_1: string;
            voie: string;
            commune: string;
            cp: number;
            pays: string;
        };
    };
}
