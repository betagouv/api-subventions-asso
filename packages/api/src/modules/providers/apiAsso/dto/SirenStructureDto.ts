export interface SirenStructureEtablissementDto {
    actif: boolean;
    adresse: {
        cplt_1?: string;
        num_voie: number;
        voie: string;
        commune: string;
        code_insee: number;
        cp: number;
        type_voie: string;
    };
    id_siret: number;
    est_siege: boolean;
    date_actif: string;
    id_activite_principale: string;
    annee_activite_principale: number;
    effectif_salarie_cent: number;
    id_tranche_effectif?: string;
    annee_effectif_salarie_cent: number;
}

export interface SirenStructureDto {
    identite: {
        nom: string;
        nom_sirene: string;
        id_siren: number;
        id_siret_siege: number;
        id_forme_juridique: number;
        date_creation_sirene: string;
        date_modif_siren: string;
        active: boolean;
        active_sirene: boolean;
        regime: string;
    };
    activites: {
        id_activite_principale: string;
        annee_activite_principale: number;
        id_tranche_effectif: string;
        effectif_salarie_cent: number;
        annee_effectif_salarie_cent: number;
        appartenance_ess: string;
    };
    coordonnees: {
        adresse_siege: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            code_insee: number;
            cp: number;
            type_voie: string;
        };
        adresse_gestion: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            code_insee: number;
            cp: number;
            type_voie: string;
        };
    };
    nbEtabsActifs: number;
    etablissement?: SirenStructureEtablissementDto[];
}
