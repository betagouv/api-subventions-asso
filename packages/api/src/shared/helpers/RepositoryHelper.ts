import UserDbo from "../../modules/user/repositories/dbo/UserDbo";

export const removeSecrets = <T extends UserDbo>(user: T) => {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const { hashPassword, jwt, ...userWithoutSecret } = user;
    return userWithoutSecret;
};
