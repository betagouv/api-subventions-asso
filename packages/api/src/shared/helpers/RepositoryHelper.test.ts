import { ObjectId } from "mongodb";
import * as RepositoryHelper from "./RepositoryHelper";
import { AgentTypeEnum } from "dto";

const USER_DBO = {
    _id: new ObjectId(1),
    email: "test22@beta.gouv.fr",
    hashPassword: "PASSWORD",
    roles: ["user"],
    signupAt: new Date(),
    jwt: { token: "TOKEN", expirateDate: new Date() },
    profileToComplete: false,
    active: false,
    agentType: AgentTypeEnum.CENTRAL_ADMIN,
    jobType: [],
};

describe("removeSecrets", () => {
    it("should remove all secret in user", () => {
        const actual = RepositoryHelper.removeSecrets(USER_DBO);
        expect(actual).toMatchSnapshot({
            _id: expect.any(ObjectId),
            signupAt: expect.any(Date),
        });
    });
});

describe("uniformizeId", () => {
    it("should uniformizeId id ", () => {
        const actual = RepositoryHelper.uniformizeId(USER_DBO);
        expect(actual).toMatchObject({
            _id: expect.any(String),
        });
    });
});
