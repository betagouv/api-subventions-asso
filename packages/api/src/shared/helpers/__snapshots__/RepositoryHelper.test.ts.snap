// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeSecrets should remove all secret in user 1`] = `
Object {
  "_id": Any<ObjectId>,
  "active": false,
  "email": "test22@beta.gouv.fr",
  "profileToComplete": false,
  "roles": Array [
    "user",
  ],
  "signupAt": Any<Date>,
}
`;
